TST, CI: GitHub Actions MacOS Failures

GitHub Actions MacOS CI jobs have been failing consistently. A sample log is available here: https://github.com/scipy/scipy/pull/13094/checks?check_run_id=1436142836

SciPy itself is built successfully, but runtime access to gfortran dynamic libraries (via openblas lib) is compromised, causing a cascade of failures during the testsuite that look like this:

```
E   ImportError: dlopen(/Users/runner/work/scipy/scipy/build/testenv/lib/python3.8/site-packages/scipy/spatial/qhull.cpython-38-darwin.so, 2): Library not loaded: /libgfortran.3.dylib
E     Referenced from: /Users/runner/work/scipy/scipy/openblas/lib/libopenblasp-r0.3.9.dylib
E     Reason: image not found
_ ERROR collecting build/testenv/lib/python3.8/site-packages/scipy/cluster/tests/test_hierarchy.py _
ImportError while importing test module '/Users/runner/work/scipy/scipy/build/testenv/lib/python3.8/site-packages/scipy/cluster/tests/test_hierarchy.py'.
```

There are quite a few shims in that CI code that deal with dynamic library behavior--I'll try to check this today as it will be important for CI to be consistently green with 1.6.0 branching imminent. If someone else is already close to a solution please chime in to avoid duplicated effort.

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -14,7 +14,7 @@ on:
 jobs:
   test_macos:
 
-    if: "github.repository == 'scipy/scipy' && !contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[skip github]')"
+    if: "github.repository == 'tylerjereddy/scipy' && !contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[skip github]')"
     runs-on: macos-latest
     strategy:
       max-parallel: 3
@@ -69,9 +69,11 @@ jobs:
         # Also make it use the current install location for libgfortran, libquadmath, and libgcc_s.
         pushd openblas/lib
         install_name_tool -id $PWD/libopenblasp-r*.dylib libopenblas.dylib
-        install_name_tool -change /usr/local/gfortran/lib/libgfortran.3.dylib `$FC -v 2>&1 | perl -nle 'print $1 if m{--libdir=([^\s]+)}'`/libgfortran.3.dylib libopenblas.dylib
-        install_name_tool -change /usr/local/gfortran/lib/libquadmath.0.dylib `$FC -v 2>&1 | perl -nle 'print $1 if m{--libdir=([^\s]+)}'`/libquadmath.0.dylib libopenblas.dylib
-        install_name_tool -change /usr/local/gfortran/lib/libgcc_s.1.dylib `$FC -v 2>&1 | perl -nle 'print $1 if m{--libdir=([^\s]+)}'`/libgcc_s.1.dylib libopenblas.dylib
+        install_name_tool -change /usr/local/gfortran/lib/libgfortran.3.dylib /usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6/libgfortran.3.dylib libopenblas.dylib
+        install_name_tool -change /usr/local/gfortran/lib/libquadmath.0.dylib /usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6/libquadmath.0.dylib libopenblas.dylib
+        install_name_tool -change /usr/local/gfortran/lib/libgcc_s.1.dylib /usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6/libgcc_s.1.dylib libopenblas.dylib
+        ls -tlrh
+        otool -L libopenblasp-r*.dylib
         popd
         echo "[openblas]" > site.cfg
         echo "libraries = openblas" >> site.cfg
@@ -82,7 +84,7 @@ jobs:
         rm -rf /usr/local/Cellar/gcc/9.2.0_2
         #
         export PATH="$PATH:$PWD/openblas"
-
+        ls -tlrh /usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6
     - name: Install packages
       run: |
         pip install ${{ matrix.numpy-version }}
@@ -90,4 +92,5 @@ jobs:
 
     - name: Test SciPy
       run: |
+        export DYLD_LIBRARY_PATH=/usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6/:$DYLD_LIBRARY_PATH
         python -u runtests.py

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -14,7 +14,7 @@ on:
 jobs:
   test_macos:
 
-    if: "github.repository == 'tylerjereddy/scipy' && !contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[skip github]')"
+    if: "github.repository == 'scipy/scipy' && !contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[skip github]')"
     runs-on: macos-latest
     strategy:
       max-parallel: 3
@@ -72,8 +72,6 @@ jobs:
         install_name_tool -change /usr/local/gfortran/lib/libgfortran.3.dylib /usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6/libgfortran.3.dylib libopenblas.dylib
         install_name_tool -change /usr/local/gfortran/lib/libquadmath.0.dylib /usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6/libquadmath.0.dylib libopenblas.dylib
         install_name_tool -change /usr/local/gfortran/lib/libgcc_s.1.dylib /usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6/libgcc_s.1.dylib libopenblas.dylib
-        ls -tlrh
-        otool -L libopenblasp-r*.dylib
         popd
         echo "[openblas]" > site.cfg
         echo "libraries = openblas" >> site.cfg
@@ -84,7 +82,6 @@ jobs:
         rm -rf /usr/local/Cellar/gcc/9.2.0_2
         #
         export PATH="$PATH:$PWD/openblas"
-        ls -tlrh /usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6
     - name: Install packages
       run: |
         pip install ${{ matrix.numpy-version }}

